portfolio-project-matching-app/pages/browseProjects.js

```diff
@@ -23,23 +23,26 @@ const browseProjects = () => {
     // array of all technologies
     const [allTechnologies, setAllTechnologies] = useState([])
 
+    // helps to determine whether "clear search" button is visible
     const [isSearching, setIsSearching] = useState(false)
 
     useEffect(() => {
-        initializeProjects()
-        initializeTechnologies()
+        // tracks whether component mounted, cleanup will assign false
+        let isMounted = true
+        // get projects and set state if component mounted
+        getAllProjects().then((projects) => {
+            if (isMounted) setVisibleProjects(projects)
+        })
+        // get technologies and set state if component mounted
+        getAllTechnologies().then((technologies) => {
+            if (isMounted) setAllTechnologies(technologies)
+        })
+        // cleanup function to assign false to isMounted
+        return function cleanup() {
+            isMounted = false
+        }
```

Comment:
Per Josh's suggestion, added cleanup function to indicate if the component did not mount. In the case the component did not end up mounting, we do NOT try to set our state.